scripts/slack_notification_action.go

```diff
@@ -0,0 +1,246 @@
+						Short: true,
+					},
+				}
+				mainFields = append(field, mainFields...)
+			case "commit":
+				field := []Field{
+					{
+						Title: "Commit",
+						Value: "<" + os.Getenv("GITHUB_SERVER_URL") + "/" + os.Getenv("GITHUB_REPOSITORY") + "/commit/" + os.Getenv("GITHUB_SHA") + "|" + commit_sha + ">",
+						Short: true,
+					},
+				}
+				mainFields = append(field, mainFields...)
+			}
+		}
+		fields = append(mainFields, fields...)
+	} else {
+		mainFields := []Field{
+			{
+				Title: "Ref",
+				Value: os.Getenv("GITHUB_REF"),
+				Short: true,
+			}, {
+				Title: "Event",
+				Value: os.Getenv("GITHUB_EVENT_NAME"),
+				Short: true,
+			},
+			{
+				Title: "Actions URL",
+				Value: "<" + os.Getenv("GITHUB_SERVER_URL") + "/" + os.Getenv("GITHUB_REPOSITORY") + "/commit/" + os.Getenv("GITHUB_SHA") + "/checks|" + os.Getenv("GITHUB_WORKFLOW") + ">",
+				Short: true,
+			},
+			{
+				Title: "Commit",
+				Value: "<" + os.Getenv("GITHUB_SERVER_URL") + "/" + os.Getenv("GITHUB_REPOSITORY") + "/commit/" + os.Getenv("GITHUB_SHA") + "|" + commit_sha + ">",
+				Short: true,
+			},
+			{
+				Title: os.Getenv(EnvSlackTitle),
+				Value: envOr(EnvSlackMessage, "EOM"),
+				Short: false,
+			},
+		}
+		fields = append(mainFields, fields...)
+	}
+
+	hostName := os.Getenv(EnvHostName)
+	if hostName != "" {
+		newfields := []Field{
+			{
+				Title: os.Getenv("SITE_TITLE"),
+				Value: os.Getenv(EnvSiteName),
+				Short: true,
+			},
+			{
+				Title: os.Getenv("HOST_TITLE"),
+				Value: os.Getenv(EnvHostName),
+				Short: true,
+			},
+		}
+		fields = append(newfields, fields...)
+	}
+
+	color := ""
+	switch os.Getenv(EnvSlackColor) {
+	case "success":
+		color = "good"
+	case "cancelled":
+		color = "#808080"
+	case "failure":
+		color = "danger"
+	default:
+		color = envOr(EnvSlackColor, "good")
+	}
+
+	msg := Webhook{
+		UserName:  os.Getenv(EnvSlackUserName),
+		IconURL:   os.Getenv(EnvSlackIcon),
+		IconEmoji: os.Getenv(EnvSlackIconEmoji),
+		Channel:   os.Getenv(EnvSlackChannel),
+		LinkNames: os.Getenv(EnvSlackLinkNames),
+		Attachments: []Attachment{
+			{
+				Fallback:   envOr(EnvSlackMessage, "GITHUB_ACTION="+os.Getenv("GITHUB_ACTION")+" \n GITHUB_ACTOR="+os.Getenv("GITHUB_ACTOR")+" \n GITHUB_EVENT_NAME="+os.Getenv("GITHUB_EVENT_NAME")+" \n GITHUB_REF="+os.Getenv("GITHUB_REF")+" \n GITHUB_REPOSITORY="+os.Getenv("GITHUB_REPOSITORY")+" \n GITHUB_WORKFLOW="+os.Getenv("GITHUB_WORKFLOW")),
+				Color:      color,
+				AuthorName: envOr(EnvGithubActor, ""),
+				AuthorLink: os.Getenv("GITHUB_SERVER_URL") + "/" + os.Getenv(EnvGithubActor),
+				AuthorIcon: os.Getenv("GITHUB_SERVER_URL") + "/" + os.Getenv(EnvGithubActor) + ".png?size=32",
+				Footer:     envOr(EnvSlackFooter, "<https://github.com/rtCamp/github-actions-library|Powered By rtCamp's GitHub Actions Library>"),
```

Comment:
Can we do this without a footer? If not, can we make the footer empty? I don't see a benefit in advertising/adding footers here.

Reply:
Yeah, happy to leave this empty.